### PR TITLE
ACA-51: Implement "Unassign Access within Organisation" Operation

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [ACA-51](https://tools.hmcts.net/jira/browse/ACA-51) _'Implement "Unassign Access within Organisation" Operation'_

### Change description ###

Create endpoint for Unassign Access within Organisation

> Note: Just a DRAFT PR as this is currently against ACA-59 as it conatins a merge of ACA-59

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature
Comment (mattnayler) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature:229]: This AC is obsolete and has been removed.

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature
Comment (mattnayler) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature:248]: This AC is obsolete and has been removed.

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203_Prerequisite_Case_Assignment_Seperate_Org_Policies_C1_Benjamin_By_Becky.td.json
Comment (mattnayler) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203_Prerequisite_Case_Assignment_Seperate_Org_Policies_C1_Benjamin_By_Becky.td.json:26]: This Tab/Space issue should now be resolved

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203_Prerequisite_Case_Assignment_Seperate_Org_Policies_C1_Other_Org2_By_CAA_Org2.td.json
Comment (mattnayler) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203_Prerequisite_Case_Assignment_Seperate_Org_Policies_C1_Other_Org2_By_CAA_Org2.td.json:26]: This step is obsolete and has been removed.

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature
Comment (mattnayler) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature:170]: This AC has been updated to remove these steps which cannot be achieved using current wiremock.

File: src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java:142]: Better split this into a few methods with nicely self-descriptive names each.

File: src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/AuthError.java
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/AuthError.java]: Agreed with @mattnayler that this is a standard error response

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/S-203.8.td.json
Comment (sullivangareth) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/S-203.8.td.json]: @MSancaktutar What are your thoughts on this - acceptance criteria states

"Must return an error response if the invoker is a solicitor in a different jurisdiction from that of the case."

File: build.gradle
Comment (MSancaktutar) [reply build.gradle:268]: Yes.

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Comment by sullivangareth** [comment]
This ACA-51 branch will need to be merged into `master`

Wait until `ACA-59` has been merged into `master` before merging this PR

File: src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature
Comment (sullivangareth) [reply src/functionalTest/resources/features/F-203 - Unassign Access Within Organisation/F-203 - Unassign Access Within Organisation.feature:114]: Happy with explanation that R1 and R2 are the roles.

**Pull request merged**
